### ISSUE TITLE
OTP setup results in 500 Error

### ISSUE THREAD

**Issue by Zebble** [PowerDNS-Admin/PowerDNS-Admin#1552]
I'm having this same issue as #1403.

I'm using LDAP (AD) as the external authentication provider.  When I force OTP, I get the OTP setup screen with the QR code.  When I scan it and enter the code, I get a 500 error and the tracebacks below.

One note that might help is that when the OTP setup screen pops up, "None" is displayed in the "Your secret key is:" box, instead of the key.  There is no debug error in the logs prior or after the OTP setup screen is shown, only after the code is first entered.

Tracebacks:

Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python3.10/site-packages/flask/app.py", line 1519, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3.10/site-packages/flask/app.py", line 1517, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3.10/site-packages/flask/app.py", line 1503, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/app/powerdnsadmin/routes/index.py", line 864, in welcome
    good_token = user.verify_totp(otp_token)
  File "/app/powerdnsadmin/models/user.py", line 99, in verify_totp
    return totp.verify(token, valid_window = 5)
  File "/usr/lib/python3.10/site-packages/pyotp/totp.py", line 80, in verify
    if utils.strings_equal(str(otp), str(self.at(for_time, i))):
  File "/usr/lib/python3.10/site-packages/pyotp/totp.py", line 56, in at
    return self.generate_otp(self.timecode(for_time) + counter_offset)
  File "/usr/lib/python3.10/site-packages/pyotp/otp.py", line 35, in generate_otp
    hasher = hmac.new(self.byte_secret(), self.int_to_bytestring(input), self.digest)
  File "/usr/lib/python3.10/site-packages/pyotp/otp.py", line 49, in byte_secret
    missing_padding = len(secret) % 8
TypeError: object of type 'NoneType' has no len()
[2023-04-27 13:12:56,788] [app.py:1449] ERROR - Exception on /welcome [POST]
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python3.10/site-packages/flask/app.py", line 1519, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3.10/site-packages/flask/app.py", line 1517, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3.10/site-packages/flask/app.py", line 1503, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/app/powerdnsadmin/routes/index.py", line 864, in welcome
    good_token = user.verify_totp(otp_token)
  File "/app/powerdnsadmin/models/user.py", line 99, in verify_totp
    return totp.verify(token, valid_window = 5)
  File "/usr/lib/python3.10/site-packages/pyotp/totp.py", line 80, in verify
    if utils.strings_equal(str(otp), str(self.at(for_time, i))):
  File "/usr/lib/python3.10/site-packages/pyotp/totp.py", line 56, in at
    return self.generate_otp(self.timecode(for_time) + counter_offset)
  File "/usr/lib/python3.10/site-packages/pyotp/otp.py", line 35, in generate_otp
    hasher = hmac.new(self.byte_secret(), self.int_to_bytestring(input), self.digest)
  File "/usr/lib/python3.10/site-packages/pyotp/otp.py", line 49, in byte_secret
    missing_padding = len(secret) % 8
TypeError: object of type 'NoneType' has no len()

_Originally posted by @Zebble in https://github.com/PowerDNS-Admin/PowerDNS-Admin/issues/1403#issuecomment-1525681941_

**Comment by AzorianMatt** [comment]
@Zebble how did you post this issue without going through the issue templates?

**Comment by Zebble** [comment]
I used the option the open as a new issue in issue #1403.  Since that one was closed, this seemed to make sense.
